File: Data Structures & Algorithms/burst-balloons/submission-0.py
class Solution:
    def maxCoins(self, nums: List[int]) -> int:
        arr = [1] + nums + [1]
        n = len(arr)
        dp = [[0]*n for _ in range(n)]

        for length in range(2,n):
            for l in range(0,n-length):
                r = l+length
                best= 0
                for k in range(l+1,r):
                    best = max(best,arr[l]*arr[k]*arr[r]+dp[l][k]+dp[k][r])
                dp[l][r] = best
        return dp[0][n-1]

File: Data Structures & Algorithms/majority-element/submission-0.java
class Solution {
    public int majorityElement(int[] nums) {
        // boyer moore Algo - candidate, vote.
        // initially candidate = -1, vote = 0;
        // traverse the array; if vote==0, set candidate = arr[i] and vote to 1.
        // if arr[i]=candidate, vote++;
        // else, vote--;

        // count occurances of candidate, if >n/2; return that else -1.

        int candidate=-1;
        int vote=0;
        for(int i=0;i<nums.length;i++){
            if(vote==0){
                candidate = nums[i];
            }
            if(nums[i]==candidate){
                vote++;
            }
            else{
                vote--;
            }
        }
        if(vote>0){
            return candidate;
        }
        return -1;
    }
}